main.js

```diff
@@ -693,33 +691,35 @@ class Sourceanalytix extends utils.Adapter {
 
 			default:
 				this.log.error("Error in case handling of cost type identificaton : " + obj_cust.state_type);
-				skip_calc = true;
+				return;
 		}
 
 		// Disable initalisatiton in case of w value is received
-		if (this.defineUnit(obj_cont) === "w"){skip_calc === true;}
+		if (this.defineUnit(obj_cont) === "w") return;
 
-		if (skip_calc === false){
+		// Get current value from meter
+		const reading = await this.getForeignStateAsync(id._id);
+		if (!reading) {
+			// maybe log something?
```

Comment:
Its best to stop executing the function here - not sure if you want to log something.